### PR TITLE
lis2ds12 accelerometer alignment

### PR DESCRIPTION
**Pull request by avisconti**
Improvements:                                                                                
                                                                                
0. Aligned stmesc API to latest DT facilities                                                                               
1. Make use of latest stmemsc module lis2ds12_STdC APIs                                                                                
2. Add multi-instance support                                                                                
3. Move range and odr into DT                                                                                
                                                                                
Tested with x_nucleo_iks01a3 shield in SPI mode

### CODE REVIEW

File: drivers/sensor/lis2ds12/lis2ds12_spi.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12_spi.c]: nit: extra space after `)`

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12.c:372]: `SPI_DT_SPEC_INST_GET` should be used

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12.c:399]: `I2C_DT_SPEC_INST_GET`

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12_trigger.c:27]: this function should be checked for errors

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12_trigger.c:58]: ditto

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12_trigger.c:163]: ditto

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12_trigger.c:185]: ditto

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (gmarull) [inline drivers/sensor/lis2ds12/lis2ds12_trigger.c:200]: ditto

**Review by gmarull — CHANGES_REQUESTED** [review]
Changes lgtm, some minor comments that I think should be addressed

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12.c:372]: It seems fine, as it is a code semplification. But I have to change it into stmemsc_cfg_spi structure, which is common to many ST sensor now. So, I have to do it as a preliminary commit for those sensors first, and I will do it in this PR itself. Sounds good?

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12.c]: ack

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12.c:92]: ack

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12.c:107]: ack

File: drivers/sensor/lis2ds12/lis2ds12_spi.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_spi.c]: ack

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:27]: But the callback returns void in this case. Do you want me to raise a LOG_ERR here?

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:58]: same here. LOG_ERR?

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:163]: ack

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:185]: ack

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:200]: ack

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (gmarull) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:27]: Maybe an assert then?

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (gmarull) [reply drivers/sensor/lis2ds12/lis2ds12.c:372]: It’s ok, just raised as it’s always better to use existing facilities when available.

File: drivers/sensor/lis2ds12/lis2ds12_trigger.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12_trigger.c:27]: sure

File: drivers/sensor/lis2ds12/lis2ds12.c
Comment (avisconti) [reply drivers/sensor/lis2ds12/lis2ds12.c]: > It’s ok, just raised as it’s always better to use existing facilities when available.

Yes, correct. THanks for pointing this out. The stmemsc common APIs are now quite simpler.

**Review by gmarull — APPROVED** [review]

**Comment by jefffhaynes** [comment]
@MaureenHelm @mbolivar-nordic I could really use this merge.  Any chance of getting some reviews?

**Comment by avisconti** [comment]
> @MaureenHelm @mbolivar-nordic I could really use this merge. Any chance of getting some reviews?

@jefffhaynes I think this PR was blocked by v2.7.0 release.

**Review by MaureenHelm — COMMENTED** [review]
LGTM.

@avisconti will you please rebase? I'd like to see a fresh CI run on this PR before merging.

**Comment by avisconti** [comment]
Still running. Mmmmh...

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
